domain/consensus/finality_test.go

```diff
@@ -0,0 +1,164 @@
+				t.Fatalf("Block #%d in main chain expected to have status '%s', but got '%s'",
+					i, externalapi.StatusValid, blockInfo.BlockStatus)
+			}
+		}
+
+		// Mine another chain of `finality-Interval - 2` blocks
+		var sideChainTip *externalapi.DomainBlock
+		sideChainTipHash := params.GenesisHash
+		for i := uint64(0); i < finalityInterval-2; i++ {
+			sideChainTip, err = buildAndInsertBlock([]*externalapi.DomainHash{sideChainTipHash})
+			if err != nil {
+				t.Fatalf("TestFinality: Failed to process sidechain Block #%d: %v", i, err)
+			}
+			sideChainTipHash = consensusserialization.BlockHash(sideChainTip)
+
+			blockInfo, err := consensus.GetBlockInfo(sideChainTipHash)
+			if err != nil {
+				t.Fatalf("TestFinality: Block #%d failed to get info: %v", i, err)
+			}
+			if blockInfo.BlockStatus != externalapi.StatusUTXOPendingVerification {
+				t.Fatalf("Block #%d in side chain expected to have status '%s', but got '%s'",
+					i, externalapi.StatusUTXOPendingVerification, blockInfo.BlockStatus)
+			}
+		}
+
+		// Add two more blocks in the side-chain until it becomes the selected chain
+		for i := uint64(0); i < 2; i++ {
+			sideChainTip, err = buildAndInsertBlock([]*externalapi.DomainHash{sideChainTipHash})
+			if err != nil {
+				t.Fatalf("TestFinality: Failed to process sidechain Block #%d: %v", i, err)
+			}
+			sideChainTipHash = consensusserialization.BlockHash(sideChainTip)
+		}
+
+		// Make sure that now the sideChainTip is valid and selectedTip
+		blockInfo, err := consensus.GetBlockInfo(sideChainTipHash)
+		if err != nil {
+			t.Fatalf("TestFinality: Failed to get block info: %v", err)
+		}
+		if blockInfo.BlockStatus != externalapi.StatusValid {
+			t.Fatalf("TestFinality: Overtaking block in side-chain expected to have status '%s', but got '%s'",
+				externalapi.StatusValid, blockInfo.BlockStatus)
+		}
+		selectedTip, err := consensus.GetVirtualSelectedParent()
+		if err != nil {
+			t.Fatalf("TestFinality: Failed getting virtual selectedParent: %v", err)
+		}
+		if *consensusserialization.BlockHash(selectedTip) != *sideChainTipHash {
+			t.Fatalf("Overtaking block in side-chain is not selectedTip")
+		}
+
+		// Add two more blocks to main chain, to move finality point to first non-genesis block in mainChain
+		for i := uint64(0); i < 2; i++ {
+			mainChainTip, err = buildAndInsertBlock([]*externalapi.DomainHash{mainChainTipHash})
+			if err != nil {
+				t.Fatalf("TestFinality: Failed to process sidechain Block #%d: %v", i, err)
+			}
+			mainChainTipHash = consensusserialization.BlockHash(mainChainTip)
+		}
+
+		dagFinality, err := consensus.DAGTraversalManager().BlockAtDepth(model.VirtualBlockHash, params.FinalityDepth())
+		if err != nil {
+			t.Fatalf("TestFinality: Failed getting the virtual's finality point: %v", err)
+		}
+
+		if *dagFinality == *params.GenesisHash {
+			t.Fatalf("virtual's finalityPoint is still genesis after adding finalityInterval + 1 blocks to the main chain")
```

Comment:
This doesn't translate very well from the old test.
I realize that you've copy+pasted csm.virtualFinalityPoint() here, but the casual reader probably would not.

I suggest you either write a comment explaining this or externalize virtualFinalityPoint in the CSM.